subprojects/files/src/testFixtures/groovy/org/gradle/api/internal/file/collections/AbstractDirectoryWalkerTest.groovy

```diff
@@ -162,11 +162,8 @@ abstract class AbstractDirectoryWalkerTest<T> extends Specification {
 
         when:
         dir.deleteDir()
-        walkDirForPaths(walkerInstance, rootDir, new PatternSet())
-
         then:
-        def e = thrown Exception
-        e.message.contains("Could not list contents of '${link.absolutePath}'.")
+        walkDirForPaths(walkerInstance, rootDir, new PatternSet()) != null
```

Comment:
The title of this test has to be changed to reflect the new behavior. We should also assert what has been walked now that the code actually returns something.

Reply:
Indeed, thanks for noticing